### DISCUSSION TITLE
how to make 'answer' parameter dynamic in same-different-image?

### DISCUSSION THREAD

**Discussion by benwandrew** [jspsych/jsPsych#1585] [Q&A]
hi all, 

in a same-different-image trial, i'd like to make the value that the answer parameter takes dynamic. specifically, i'd like it to depend on whether the first_stim_duration (fixed at 500ms) is longer than the the second_stim_duration (randomly sampled from an array of 100ms up to 900ms by increments of 100ms).

is it some simple if/else logic? if so, where can i find the correct syntax for writing such a statement? 

if (first_stim_duration < second_stim_duration)... doesn't seem to be working.

i'm new to jspsych all around, so i'd really appreciate any pointers!

**Comment by jodeleeuw** [comment]
Hi @benwandrew ,

In general to make parameters dynamic you need to use functions. [Here's the relevant docs page](https://www.jspsych.org/overview/dynamic-parameters/).

In this case it sounds like you want to have one dynamic parameter that depends on two other dynamic parameters. You can do this via the `on_start` event for the trial. [Docs](https://www.jspsych.org/overview/callbacks/#on_start-trial).
